services/src/main/java/org/exoplatform/addons/gamification/service/configuration/AnnouncementServiceImpl.java

```diff
@@ -176,6 +176,34 @@ public Announcement updateAnnouncement(Announcement announcement, boolean broadc
     return savedAnnouncement;
   }
 
+  @Override
+  public Announcement cancelAnnouncement(Long announcementId, String username) throws ObjectNotFoundException,
+                                                                               IllegalAccessException {
+    if (announcementId == null || announcementId <= 0) {
+      throw new IllegalArgumentException("announcement id is mandatory");
+    }
+    Announcement announcementToCancel = announcementStorage.getAnnouncementById(announcementId);
+    if (announcementToCancel == null) {
+      throw new ObjectNotFoundException("Announcement does not exist");
+    }
+    Identity identity = identityManager.getOrCreateUserIdentity(username);
+    if (!announcementToCancel.getCreator().equals(Long.parseLong(identity.getId()))) {
+      throw new IllegalAccessException("user " + username + " is not allowed to cancel announcement with id "
+          + announcementToCancel.getId());
+    }
+    String activityId = String.valueOf(announcementToCancel.getActivityId());
+    Announcement canceledAnnouncement = announcementStorage.cancelAnnouncement(announcementToCancel);
+    try {
+      activityManager.deleteActivity(activityId);
+    } catch (Exception e) {
+      LOG.warn("Error while deleting activity for announcement with with id {} made by user {}",
+               announcementToCancel.getId(),
+               announcementToCancel.getCreator(),
+               e);
+    }
```

Comment:
Never catch exception about treatment in Service layer, handle it in REST with adequate Response code to give a good UX for users by indicating that their response fails.
Besides, it would be better to **delete the activity before deleting announcement** for better data coherence.
In addition, it would be better to hide the activity instead of deleting it IMO. (preserve data instead of losing it forever, as done for the Action History which is just rejected and not deleted)

```suggestion
      activityManager.deleteActivity(activityId);
```